src/github/handlers/push-event.ts

```diff
@@ -0,0 +1,260 @@
+        },
+        body
+      );
+    }
+  } catch (e) {
+    console.error("handleActionValidationWorkflowCompleted", e);
+  }
+}
+
+async function createCommitComment(
+  context: GitHubContext,
+  { owner, repo, commitSha, userLogin }: { owner: string; repo: string; commitSha: string; userLogin?: string },
+  body: string[]
+) {
+  const { octokit } = context;
+
+  const commit = (
+    await octokit.rest.repos.listCommentsForCommit({
+      owner: owner,
+      repo: repo,
+      commit_sha: commitSha,
+    })
+  ).data
+    .filter((o) => o.user?.type === "Bot")
+    .pop();
+  if (commit) {
+    await octokit.rest.repos.updateCommitComment({
+      owner: owner,
+      repo: repo,
+      commit_sha: commitSha,
+      comment_id: commit.id,
+      body: `${commit.body}\n${body.join("")}`,
+    });
+  } else {
+    body.unshift(`@${userLogin} Configuration is invalid.\n`);
+    await octokit.rest.repos.createCommitComment({
+      owner: owner,
+      repo: repo,
+      commit_sha: commitSha,
+      body: body.join(""),
+    });
+  }
+}
+
+async function checkPluginConfigurations(context: GitHubContext<"push">, config: PluginConfiguration, rawData: string | null) {
+  const { payload, eventHandler } = context;
+  const errors: (ValueError | YAML.YAMLError)[] = [];
+
+  for (let i = 0; i < config.plugins.length; ++i) {
+    const { uses } = config.plugins[i];
+    for (let j = 0; j < uses.length; ++j) {
+      const { plugin, with: args } = uses[j];
+      const isGithubPluginObject = isGithubPlugin(plugin);
+      const stateId = crypto.randomUUID();
+      const token = payload.installation ? await eventHandler.getToken(payload.installation.id) : "";
+      const ref = isGithubPluginObject ? (plugin.ref ?? (await getDefaultBranch(context, plugin.owner, plugin.repo))) : plugin;
+      const inputs = new PluginInput(context.eventHandler, stateId, context.key, payload, args, token, ref);
+
+      if (!isGithubPluginObject) {
+        try {
+          const response = await dispatchWorker(`${plugin}/manifest`, await inputs.getWorkerInputs());
+          const responseWithDefaults = Value.Default(pluginValidationResponseSchema, response) as StateValidation;
+          if (!stateValidationErrorSchemaValidator.test(responseWithDefaults)) {
+            console.error("Malformed response from the endpoints");
+            for (const err of stateValidationErrorSchemaValidator.errors(responseWithDefaults)) {
+              console.error(err);
+            }
+          }
+          const decodedResponse = Value.Decode(pluginValidationResponseSchema, responseWithDefaults);
+          if (decodedResponse.errors.length) {
+            errors.push(...decodedResponse.errors.map((err) => ({ ...err, path: `plugins/${i}/uses/${j}/with${err.path}` })));
+          }
+        } catch (e) {
+          errors.push({
+            path: `plugins/${i}/uses/${j}`,
+            message: `Failed to reach plugin endpoint: ${e}`,
+            value: plugin,
```

Comment:
here you didn't stringify the plugin like below, I'm not sure which is correct

Reply:
It's because our plugin is either a string (url) or an object (repo, ref, etc.) so if I don't stringify the other one we get "[object]" in the output instead of the github repo.